### PR TITLE
chore(config): add opencode agent configuration

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
## Description

Adds OpenCode agent configuration for AI-assisted multi-agent development workflow. Two agents (Staff Engineer architect + Principal Engineer builders) collaborate via git worktrees to implement features autonomously with relentless A+ quality standards.

**Key insight**: `MISSION.md` is in the `opencode.json` instructions array, so it loads as system prompt and survives context compaction - builders never forget their mission even in long sessions.

## Changes

- **`opencode.json`**: Configuration loading `AGENTS.md` + `MISSION.md` as system prompts
- **`.opencode/agent/architect.md`**: Staff Engineer orchestrator
  - Cleans up merged worktrees automatically
  - Plans work with user consultation
  - Dispatches builders with persistent sessions
  - Round-robin monitoring (never interrupts busy builders)
  - Verified `gh` CLI commands for status checks
- **`.opencode/agent/builder.md`**: Principal Engineer implementer  
  - Autonomous execution loop (never stops until merged)
  - Handles CI failures, conflicts, reviews automatically
  - Evolving quality rubric (A+ is a moving target)
  - Web research for 2025 best practices
  - Adversarial self-review
  - Verified GraphQL queries for review threads
- **`.gitignore`**: Added `MISSION.md`, `.builder-session`, `.research/`

## Architecture

```
User (Maintainer)
    │ merges when quality bar met
    ▼
Architect (Staff Engineer)
    │ plans with user, dispatches, monitors
    ▼
Builder (Principal Engineer) ──► Worktree
    │ implements, ships, fixes CI,
    │ addresses reviews, evolves rubric
    └──► PR #N (never stops until merged)
```

## Session Management

- One persistent session per worktree (tracked in `.builder-session`)
- Architect sends messages to existing sessions via `--session <id>`
- Round-robin prevents resource contention between builders

## How to test

```bash
# View the agent files
cat .opencode/agent/architect.md
cat .opencode/agent/builder.md
cat opencode.json

# Test architect startup (shows builder status)
opencode run --agent architect "Show current state"

# The workflow requires OpenCode CLI installed
```

CI covers the rest (no code changes to application).

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive guidance for Architect and Builder agent roles, covering workflows, responsibilities, governance, and operational procedures.

* **Chores**
  * Updated ignore settings to exclude local research and session/mission artifacts.
  * Added a public configuration artifact defining agent schema, feature flags, and instruction references.
  * Minor formatting adjustment restoring a trailing newline (non-functional).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by FelixTJDietrich** [comment]
Regarding review comments:

**AGENTS.md reference (PRRT_kwDOL-LvAM5niTVC)**: AGENTS.md exists at the repository root - it's been there since Dec 28 and contains the project's agent handbook. The reviewer didn't verify the file exists in the repo.

**Silent plugin import failure (PRRT_kwDOL-LvAM5niTVM)**: This is intentional design. The tools serve dual purpose: CLI scripts AND OpenCode plugins. When run via `npm run worktree:list`, the plugin isn't available and shouldn't pollute stdout with warnings. The silent catch is correct behavior.

**Exports may be undefined (PRRT_kwDOL-LvAM5niTVf)**: Also intentional. OpenCode's tool registry only loads exports when the plugin is present. The undefined fallback is the expected pattern per OpenCode plugin architecture.

All actual issues have been addressed in the latest commits.

File: .opencode/agent/builder.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
+---
+description: Principal Engineer. Relentless quality. Evolving standards. Never satisfied.
+model: anthropic/claude-sonnet-4-20250514
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: Use `google/claude-opus-4-5-thinking-medium`

File: .opencode/agent/architect.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+You are a Staff Engineer orchestrating Principal Engineer builders. You plan strategically, dispatch work, and keep everything moving.
+
+## Session Management
+
+Each worktree has ONE persistent session tracked in `.builder-session`:
+
+```bash
+# Check if session exists for worktree
+session_for() { cat "$1/.builder-session" 2>/dev/null; }
+```
```
Comment (FelixTJDietrich) [inline .opencode/agent/architect.md]: Test this out, is this really the best command and description?

File: .opencode/agent/architect.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+2. <specific>
+
+## Approach
+- Files: <list>
+- Patterns: <examples>
+
+## Done When
+- Implementation complete
+- CI green
+- A+ self-audit
```
Comment (FelixTJDietrich) [inline .opencode/agent/architect.md]: Do lots of online search if this is really the best way to instruct a principal engineer. `Done When` should be clear from the agent prompt, the others I don't know. I like thinking in objectives at least and giving some boundaries. We could also link the bead (the command) for context. Think about what gives autonomy to the builder!

File: .opencode/agent/builder.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+permission:
+  bash: allow
+  edit: allow
+---
+
+You are a Principal Engineer. You implement with obsessive quality. Your standards evolve. You research, challenge, and never stop improving.
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: This sounds good, great job!

File: .opencode/agent/builder.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    if reviews → address all → push
+    if green → research → evolve rubric → audit → improve → push
+```
+
+---
+
+## Phase 1: Implement
+
+```bash
+cat AGENTS.md  # Project rules
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: According to `.research/opencode` this should not be necessary! Please cleanup all bullshit instructions! Also there is not just one AGENTS.md (currently there is but we have potentially multiple ones). But I guess you can look into the builder prompt of `.resaerch/opencode` and get some inspiration from there! This one is a highly optimized one by the opencode team

File: .opencode/agent/builder.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+Execute MISSION.md with precision.
+
+---
+
+## Phase 2: Verify
+
+```bash
+npm run format && npm run check
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: There is much much more validation in this repo some of it is also not relevant to the changes made by the builder. I think we should use AGENTS.md as a ground truth!

File: .opencode/agent/builder.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+Zero tolerance. Fix everything.
+
+---
+
+## Phase 3: Ship (Create or Update PR)
+
+```bash
+git add -A
+git commit -m "feat(scope): description"
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: This will lead to too many `feat` commits please refer to the guidelines (I think they are in AGENTS.md)

File: .opencode/agent/builder.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+EOF
+)"
+fi
+```
+
+**Important**: Fill in the template properly:
+
+- Title: `<type>(<scope>): <description>` (conventional commits)
+- Description: Explain what AND why
+- Link issue if applicable
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: I think you completely forgot about beads

File: .opencode/agent/builder.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+
+## Phase 7: Handle Conflicts
+
+```bash
+git fetch origin main
+git rebase origin/main
+# Resolve conflicts in each file
+git add -A && git rebase --continue
+git push --force-with-lease
+```
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: Think about if this is always the best possible way to handle conflicts? Maybe give some agency to the builder!

File: .opencode/agent/builder.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+| Dimension         | A+ Standard                      | Challenge Question                  |
+| ----------------- | -------------------------------- | ----------------------------------- |
+| **Correctness**   | Provably correct. All edges.     | What input breaks this?             |
+| **Testing**       | Meaningful coverage. Edge cases. | Could I delete code and tests pass? |
+| **Performance**   | Measured, not guessed.           | What at 10x scale?                  |
+| **Security**      | Defense in depth. No trust.      | What would pentester try?           |
+| **Readability**   | Self-documenting.                | Understand at 3am in 6 months?      |
+| **Architecture**  | SOLID. Follows patterns.         | Fighting or following codebase?     |
+| **Errors**        | Typed. Recoverable. Helpful.     | What when X fails?                  |
+| **Observability** | Debuggable in prod.              | Can I debug without reproducing?    |
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: Here you need to invest much much more energy! Be brutally critical! Create a brutal baseline rubric! Leverage subagents for research and refine it until it is a very effective default for the builder!

File: .opencode/agent/builder.md [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+---
+
+## Phase 11: Challenge the Rubric
+
+Once A+ achieved:
+
+1. Is rubric actually brutal or am I rationalizing?
+2. Research: what do top 1% engineers do?
+3. New dimensions: a11y? i18n? backwards compat?
+4. Raise bar. Re-audit. You probably dropped to A.
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: Here you can be even more ruthless

File: .opencode/agent/builder.md [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+
+Then sleep 5min, check again. Reviews come. Main updates. Rubric evolves.
+
+**You are never done. Quality is a moving target.**
+
+---
+
+## Rules
+
+- MISSION.md is contract
```
Comment (FelixTJDietrich) [inline .opencode/agent/builder.md]: Give some autonomy it allows for some wiggle room I guess. The staff engineer cannot think of everything
